internal/storage/postgres_storage.go

```diff
@@ -0,0 +1,130 @@
+package storage
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/ivanlp-p/ShortLinkService/internal/models"
+	"github.com/jackc/pgx/v5"
+	"github.com/jackc/pgx/v5/pgxpool"
+)
+
+type PostgresStorage struct {
+	pool *pgxpool.Pool
+}
+
+func NewPostgresStorage(ctx context.Context, dsn string) (Storage, error) {
```

Comment:
Нет смысла возвращать здесь ссылку на интерфейс, стоит вернуть указатель на тип реализации.
```suggestion
func NewPostgresStorage(ctx context.Context, dsn string) (*PostgresStorage, error) {
```